### ISSUE TITLE
Divinity Original Sin 2 not working in Flatpak

### ISSUE THREAD

**Issue by ShalokShalom** [ValveSoftware/steam-for-linux#6740]
#### Your system information

* Steam client version (build number or date): current
* Distribution (e.g. Ubuntu): KaOS
* Opted into Steam client beta?: [Yes/No] No
* Have you checked for system updates?: [Yes/No] Yes

#### Steps for reproducing this issue:

1. Install the Steam [Flatpak](https://flathub.org/apps/details/com.valvesoftware.Steam) 
2. Try to install Divinity - Original Sins 2 
3. Realize they are trying to install it into a folder, that does not exist

#### EASY FIX

1. Simply check if the casual Steam directory exists. If not, check if the Flatpak repository exists. The path to Flatpak is 

> /home/$USER/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/common/Divinity Original Sin 2/

![Screenshot_20191203_021959](https://user-images.githubusercontent.com/6344099/70196512-346e2680-1709-11ea-8dc5-8b13b0698eed.png)

So in short, 
_~/.var/app/com.valvesoftware.Steam/.local/share/Steam_ __replaces__ the commonly used 
_~/.local/share/Steam_

I suggest that for all software as a fallback, Flatpak is the only way to get Steam running on my 64 Bit only OS and I guess there are other use cases as well. 

Thanks a lot :hugs:

**Comment by nanonyme** [comment]
So a bit of context:
Flatpak by default unless you request something else runs all applications in a sandbox so that they don't have permissions to write on top of files in your user home. From Flathub Steam point of view the real directory ~/.var/app/com.valvesoftware.Steam is actually your home directory. To make things a bit more complicated, there's also ~/.var/app/com.valvesoftware.Steam inside the sandbox which resolves to itself. By default the entire ~/.var/app/com.valvesoftware.Steam was visible in paths (eg the path to .exe) but we realized that Wine was for some users hitting some max path limit so games refused to start at all. To fix this we rewrote the enviroment a bit in wrapper so Steam actually sees the path as /home/$USER/.local/share/Steam/steamapps/common/Divinity Original Sin 2/ instead of /home/$USER/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/common/Divinity Original Sin 2/. (both are present in the sandbox) This is of course a bit confusing because the paths Steam talks about aren't in reality present if you try to explore to that path on your disk. You can from command line do flatpak run --command=bash com.valvesoftware.Steam to spawn bash shell inside the sandbox to see what it has eaten.

Now for why the game doesn't actually start, I don't know. It would be useful to know if the game started in native Steam with that same user account name. We might again be hitting path length limit if that's a Proton (Wine) game but in this case the problem should be visible with native Steam too if that's the case.

**Comment by kisak-valve** [comment]
@ShalokShalom, what version of Proton are you testing the game with? You can press the ⓘ button to the right of the Play button and it should tell you which compatibility tool is selected.

**Comment by ShalokShalom** [comment]
> @ShalokShalom, what version of Proton are you testing the game with? You can press the ⓘ button to the right of the Play button and it should tell you which compatibility tool is selected.

Steam Linux Runtime. I guess this is the current version

I installed it into /.var/ and the download still did not happen. 
Can you reproduce?

![Screenshot_20191205_195232](https://user-images.githubusercontent.com/6344099/70264617-194af780-1799-11ea-99cd-575f309219c9.png)

**Comment by kisak-valve** [comment]
The [Steam Linux Runtime](https://steamcommunity.com/app/221410/discussions/0/1638675549018366706/) is a separate compatibility tool and unrelated to Proton. The initial deployment of it is not compatible with Proton, and by extension Windows games. You'll want to tell Steam to use a Proton version.

This issue is effectively a duplicate of https://github.com/ValveSoftware/steam-runtime/issues/200.

**Comment by ShalokShalom** [comment]
I changed to Proton, this is what I got: 

https://sendvid.com/c8sbktyp

Installed it into the the default folder.
